jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```
@@ -0,0 +1,237 @@
+
+	}
+
+	private static void createMapping(JestClient jestClient, String indexName, String typeName) {
+		synchronized (CREATE_MAPPING_LOCK) {
+			try {
+				IndicesExists indicesExists = new IndicesExists.Builder(indexName).build();
+				boolean indexExists = jestClient.execute(indicesExists).isSucceeded();
+
+				if (!indexExists) {
+
+					CreateIndex createIndex = new CreateIndex.Builder(indexName).build();
+					jestClient.execute(createIndex);
+
+					PutMapping putMapping = new PutMapping.Builder(indexName, typeName,
+							"{\n" +
+									"  \"jmx-entry\": {\n" +
+									"    \"properties\": {\n" +
+									"      \"attributeName\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"key\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"metric\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"server\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"timestamp\": {\n" +
+									"        \"type\": \"date\"\n" +
+									"      },\n" +
+									"      \"resultAlias\": {\n" +
+									"        \"type\": \"string\",\n" +
+									"        \"index\": \"not_analyzed\"\n" +
+									"      },\n" +
+									"      \"value\": {\n" +
+									"        \"type\": \"float\"\n" +
+									"      }\n" +
+									"    }\n" +
+									"  }\n" +
+									"}").build();
+
+					JestResult result = jestClient.execute(putMapping);
+					if (!result.isSucceeded()) {
+						log.warn("Failed to create mapping: {}", result.getErrorMessage());
+					}
+					else {
+						log.info("Created mapping for index {}", indexName);
+					}
+				}
+			} catch (IOException e) {
```

I'm not an expert on ElasticSerach and Jest, but my guess is that if the creation of mappings fail, this output writer will be unuseable. So you should probably just let the exception bubble up and not hide the issue.